File: OpenMP/C/05 Reduction Counter.c
#include<stdio.h>
#include<stdlib.h>
#include<omp.h>

#define MAX_ITS 10000

int main(int argc, char **argv)
{

  int i, its_global;

its_global = 0;
#pragma omp parallel reduction(+:its_global)
{
#pragma omp for
  for (i = 0; i< MAX_ITS; ++i){
    its_global++;
  }
}

printf( "Counter records %i iterations\n", its_global);
}

File: OpenMP/Fortran/05 Reduction Counter.f90
PROGRAM loop_decompose

  USE omp_lib
  IMPLICIT NONE
  INTEGER, PARAMETER :: max_its = 10000
  INTEGER :: i, its_global


its_global = 0
!$OMP PARALLEL DO REDUCTION(+:its_global)
  DO i = 1, max_its
    its_global = its_global + 1
  END DO
!$OMP END PARALLEL DO

PRINT '(A,I0,A)','Counter records ', its_global, ' iterations'

END PROGRAM loop_decompose
